src/main/java/ro/secur/auth/security/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,68 @@
+package ro.secur.auth.security.filter;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.jsonwebtoken.Jwts;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import ro.secur.auth.configuration.JwtConfiguration;
+import ro.secur.auth.security.authentication.AuthenticationRequest;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.sql.Date;
+import java.time.LocalDate;
+
+import static ro.secur.auth.util.Api.LOGIN_URL;
+
+
+public class AuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+
+    private final AuthenticationManager authenticationManager;
+
+    private final JwtConfiguration jwtConfiguration;
+
+    public AuthenticationFilter(AuthenticationManager authenticationManager, JwtConfiguration jwtConfiguration) {
+        this.authenticationManager = authenticationManager;
+        this.jwtConfiguration = jwtConfiguration;
+        setFilterProcessesUrl(LOGIN_URL);
+    }
+
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response) throws AuthenticationException {
+
+        try {
+            AuthenticationRequest authenticationRequest = new ObjectMapper().readValue(request.getInputStream(), AuthenticationRequest.class);
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    authenticationRequest.getUsername(),
+                    authenticationRequest.getPassword()
+            );
+            return authenticationManager.authenticate(authentication);
+
+        } catch (IOException e) {
+            logger.info("Authentication not possible with the current credentials.");
+            // TODO SEE WHAT CAN WE CUSTOMIZE THIS ERROR. (errormapping)
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    protected void successfulAuthentication(HttpServletRequest request, HttpServletResponse response, FilterChain
+            chain, Authentication authResult) throws IOException, ServletException {
+
+        String token = Jwts.builder()
+                .setSubject(authResult.getName())
+                .claim("roles", authResult.getAuthorities())
```

Comment:
hardcoded string. I've seen in other class this string being used.

Reply:
Hardcoded temporarly in a static class till we find a better solution